CMakeLists.txt

```diff
@@ -532,6 +534,7 @@ if(FLATBUFFERS_BUILD_TESTS)
   if(NOT (MSVC AND (MSVC_VERSION LESS 1900)))
     compile_flatbuffers_schema_to_cpp(tests/monster_extra.fbs) # Test floating-point NAN/INF.
   endif()
+  compile_flatbuffers_schema_to_cpp(tests/struct_union.fbs)
```

Comment:
can you instead make use of `tests/union_vector/union_vector.fbs` which already has structs in a union?

Reply:
Yes, we can put that in `union_vector.fbs`. Though, we still need to add another union there for testing, as `Set()` is not generated to unions that contain multiple same-type member.